std/assembly/string.ts

```diff
@@ -229,17 +229,17 @@ export class String {
     return -1;
   }
 
-  lastIndexOf(searchString: String, fromIndex: i32 = 0): i32 {
+  lastIndexOf(searchString: String, fromIndex: i32 = i32.MAX_VALUE): i32 {
     assert(this !== null);
     if (searchString === null) searchString = changetype<String>("null");
     var len: isize = this.length;
     var searchLen: isize = searchString.length;
     if (!searchLen) return len;
     if (!len) return -1;
-    var start = clamp<isize>(fromIndex - searchLen, 0, len);
+    var start = clamp<isize>(fromIndex, 0, len - 1);
```

Comment:
Is this correct? Seems to me that this would result in an out of bounds `compare_memory` below.